packages/components/src/internal/components/navigation/ProductMenu.spec.tsx

```diff
@@ -14,199 +14,369 @@
-        });
-
-        const menuButton = mount(<ProductMenu model={model} />);
-        expect(menuButton.find('.menu-section').length).toBe(3);
-        expect(menuButton.find('i.fa-spinner').length).toBe(2);
-        expect(menuButton).toMatchSnapshot();
-        menuButton.unmount();
+
+        await waitForLifecycle(wrapper);
+        validate(wrapper);
+        wrapper.unmount();
     });
 
-    test('multiple sections with sectionConfigs', () => {
-        const productIds = ['testProduct2Columns'];
-
-        const sections = List<MenuSectionModel>().asMutable();
-        sections.push(
-            MenuSectionModel.create({
-                label: 'Sample Sets',
-                url: undefined,
-                items: sampleSetItems,
-                itemLimit: 2,
-                key: 'samples',
-            })
+    test('error', async () => {
+        const wrapper = mountWithAppServerContext(
+            <ProductMenu {...getDefaultProps()} error="Test Error" />,
+            getDefaultAppContext(),
+            getDefaultServerContext()
         );
-        sections.push(
-            MenuSectionModel.create({
-                label: 'Assays',
-                items: assayItems,
-                key: 'assays',
-            })
+
+        await waitForLifecycle(wrapper);
+        validate(wrapper, true);
+        wrapper.unmount();
+    });
+
+    test('showFolderMenu false', async () => {
+        const wrapper = mountWithAppServerContext(
+            <ProductMenu {...getDefaultProps()} showFolderMenu={false} />,
+            getDefaultAppContext(),
+            getDefaultServerContext()
         );
-        sections.push(yourItemsSection);
-        const model = new ProductMenuModel({
-            productIds,
-            isLoaded: true,
-            isLoading: false,
-            sections: sections.asImmutable(),
-        });
-
-        let sectionConfigs = List<Map<string, MenuSectionConfig>>().asImmutable();
-
-        const samplesSectionConfigs = Map<string, MenuSectionConfig>().set(
-            'samples',
-            new MenuSectionConfig({
-                iconCls: 'test-icon-cls',
-                showActiveJobIcon: false,
-            })
+
+        await waitForLifecycle(wrapper);
+        validate(wrapper, false, false);
+        wrapper.unmount();
+    });
+
+    test('dashboardImgURL', async () => {
+        const wrapper = mountWithAppServerContext(
+            <ProductMenu {...getDefaultProps()} />,
+            getDefaultAppContext(),
+            getDefaultServerContext()
         );
-        sectionConfigs = sectionConfigs.push(samplesSectionConfigs);
 
-        const twoSectionConfig = Map<string, MenuSectionConfig>().set(
-            'assays',
-            new MenuSectionConfig({
-                iconCls: 'test-icon-cls',
-            })
+        await waitForLifecycle(wrapper);
+        validate(wrapper);
+        expect(wrapper.find(ProductMenuSection).first().prop('dashboardImgURL')).toBe(false);
```

Comment:
The `dashboardImgURL` is really `false` here and not `Falsy`?

Reply:
yes, it is because of the implementation being `dashboardImgURL={isLast && dashboardURL}`. I could change that to `dashboardImgURL={isLast ? dashboardURL : undefined}` if you think that would be clearer.

Reply:
No big deal, though I might change the test to check for Falsey since that's what matters.

Reply:
Done